### DISCUSSION TITLE
Sonoff TH Elite THR320D and DS18B20 unable to set <current Temperature>

### DISCUSSION THREAD

**Discussion by msjfb** [arendst/Tasmota#17680] [Q&A]
I can display on the screen (1st line) the correct temperature of the sensor. The sensor and its temperature also show up on the Tasmota Web interface Main Menu. However when I enable the Thermostat(cmnd/Tasmota_Name/THERMOSTATMODESET 1) and set the sensor to local (cmnd/Tasmota_Name/SENSORINPUTSET 1), the <Current Temperature> in the main menu stays at 0.0, so in effect the thermostat does not operate.  If I set the Currrent temperature via MQTT, the value is updated and the thermostat seems to operate correctly.
I would expect the Current temperature should follow the sensor values automatically once the local sensor is enabled.

Am I missing a configuration element ?

**Comment by barbudor** [comment]
What is in your SENSOR message?
The name of the sensor in SENSOR and the define THERMOSTAT_SENSOR_NAME must match exactly

**Comment by msjfb** [comment]
![Sonoff THR320D 001](https://user-images.githubusercontent.com/16277804/211878505-e8141632-975c-4a83-a06a-5e8aaba0370d.png)
![Sonoff THR320D 002](https://user-images.githubusercontent.com/16277804/211878550-71bbbef8-57cb-46c6-8ada-733729b0e6ea.png)
![Sonoff THR320D 003](https://user-images.githubusercontent.com/16277804/211878596-2f1e8b2d-6224-4220-8605-a430c02783dc.png)

Here are 3 screenshots. DS18B20 used everywhere. Did I miss a definition ?
Could the cmnd  INPUTSWITCHUSE affect this ? I have not looked into it yet.

**Reply by barbudor** [reply]
THERMOSTAT_SENSOR_NUMBER should be 0
If not, the thermostat code append `_<n>` to the name of the sensor

**Comment by Sherkniem** [comment]
Hello, I would like to use THR320D and DS18B20 to measure the water temperature and to activate the heater if Measured Temperature < Threshold. Please, could you share how do you configure it?

Thank you!!

**Comment by lalo-uy** [comment]
In the Rules documentation ther is a termostat example.
